lib/doorkeeper/oauth/client_authentication/client_secret_post.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module OAuth
+    module ClientAuthentication
+      class ClientSecretPost
+        def self.matches_request?(request)
+          request.method.upcase === "POST" && request.request_parameters[:client_id].present? && request.request_parameters[:client_secret].present?
+        end
+
+        def self.authenticate(request)
```

Comment:
I'm not sure if `authenticate` is really the method name that's correct here.. maybe more `credentials` or `parse` or something? Maybe `call` or `handle` even?

Reply:
Could also call it `from_request` or something? Given how it's called?